listener/listener.go

```diff
@@ -37,43 +46,106 @@ func configure() {
 
 	eventsReader = kafka.NewReader(eventsConfig)
 
+	inventoryAddr := utils.GetenvOrFail("INVENTORY_ADDRESS")
+
+	config := inventory.NewConfiguration()
+	config.Debug = true
+	config.BasePath = inventoryAddr + INVENTORY_API_PREFIX
+
+	inventoryClient = inventory.NewAPIClient(config)
+
+	cfg := vmaas.NewConfiguration()
+	cfg.BasePath = utils.GetenvOrFail("VMAAS_ADDRESS") + VMAAS_API_PREFIX
+	cfg.Debug = true
+
+	vmaasClient = vmaas.NewAPIClient(cfg)
 }
 
-func shutdown() {
-	err := uploadReader.Close()
-	if err != nil {
-		utils.Log("err", err.Error()).Error("unable to shutdown Kafka reader")
-	}
-	err = eventsReader.Close()
+func shutdown(reader *kafka.Reader) {
+	err := reader.Close()
 	if err != nil {
 		utils.Log("err", err.Error()).Error("unable to shutdown Kafka reader")
 	}
-
 }
 
 func baseListener(reader *kafka.Reader, handler func(message kafka.Message)) {
+	defer shutdown(reader)
+
 	for {
 		m, err := reader.ReadMessage(context.Background())
 		if err != nil {
 			utils.Log("err", err.Error()).Error("unable to read message from Kafka reader")
 			panic(err)
+		}
+		// Spawn handler, not blocking the receiving thread
+		go handler(m)
+	}
+}
+
+func logHandler(m kafka.Message) {
+	utils.Log().Info("Received message [", m.Topic, "] ", string(m.Value))
+}
 
+func uploadHandler(m kafka.Message) {
+	var event PlatformEvent
+	utils.Log("msg", string(m.Value)).Info("Msg received")
+
+	err := json.Unmarshal(m.Value, &event)
+	if err != nil {
+		utils.Log("err", err.Error()).Error("Could not deserialize host event")
+		return
+	}
+	// We need the b64 identity in order to call the inventory
+	if event.B64Identity != nil && event.Url != nil {
```

Comment:
Better to use standard go style here. That means check invalid states and handle it if needed. And keep expected routine on the left.
~~~go
step1()
if check1(): {
 log error; return
}
step2()
if check2(): {
 log error; return
}
step3()
...
~~~

Reply:
Fixed